vite.config.js

```diff
@@ -0,0 +1,24 @@
+import { defineConfig, loadEnv } from "vite";
+import vue from "@vitejs/plugin-vue";
+import path from "path";
+import { resolveConfig }  from './lib/tailwindConfigUtils'
+
+const resolveConfigPath = (configPath) =>
+  path.resolve(process.cwd(), configPath);
+
+export default async ({ mode }) => {
+  process.env = { ...process.env, ...loadEnv(mode, process.cwd()) };
+  const conf = await import(resolveConfigPath(process.env.VITE_TCV_CONFIG))
+
+  return defineConfig({
+    plugins: [vue()],
+    define: {
+      __TAILWIND_CONFIG__: JSON.stringify(resolveConfig(conf.default)),
```

Comment:
Using vite to pass in a stringified version of the config instead of running a separate `koa` server with the `config.json` endpoint is one of the bigger changes here. This impacts not only how we get the config in `Canvas.vue`, but the shape of the build as well. Anytime we run a build or run the preview server, the tailwind config will be included in the JS bundle.